72_config_and_simplify_tls.md

```diff
@@ -0,0 +1,367 @@
+* `registrar_ca_cert`: Removed. Add the CA used by the registrar to
+  `trusted_ca_out` instead.
+* `registrar_my_cert`: Removed. Use `client_cert` and add the CA of the
+  tenant/verifier to `trusted_ca_in` in the registrar.
+* `registrar_private_key` Removed. Use `client_key` instead.
+* `registrar_private_key_pw`: Removed.
+* `rsa_keyname`: (Agent only) Removed. Use `server_key` instead.
+* `mtls_cert`: (Agent only) Removed. Use `server_cert` instead.
+* `keylime_ca`: (Agent only) Removed. Use `trusted_ca_in` instead. 
+* `mtls_cert_enabled`: Renamed to `enable_agent_mtls`.
+
+### Special keywords
+* `default`: If an option is set to default it will use the default of the
+  verifier as before
+  * `tls_dir`: `/var/lib/keylime/cv_ca` for server components. `/var/lib/keylime/secure` for the agent.
+  * `client_key`: `private.key` (Use generic name instead of FQDN which is more
+    portable)
+  * `client_cert`: `cert.pem` (PEM encoded)
+  * `server_key`: `private.key`
+  * `server_cert`: `cert.pem`
+  * `trusted_ca_in`: `[cacert.pem]`
+  * `trusted_ca_out`: `[cacert.pem]`
+* `all` (only in `trusted_ca_in` and `trusted_ca_out`): Disables validation of
+  the certificates.
+* `generate` (only in `tls_dir`): Generates a CA with the correct certificates.
+  Agent will generate the key and certificate if not present regardless of this option.
+
+
+### Component specific details
+* Agent: Will not have the `client_key` or `client_cert` option, because it does not connect anywhere with mTLS.
+* Tenant: The `server_key` and `server_cert` option will be only used by the tenant webapp.
+
+### User Stories (optional)
+
+<!--
+Detail the things that people will be able to do if this enhancement is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (optional)
+
+The TLS setup is moved from a connection basis to a component basis.
+
+### Risks and Mitigations
+
+Setting up the client TLS verification is important for the overall security of
+Keylime. Warning for obvious insecure settings should be added where not already
+in place. The documentation will describe the default configuration and one more
+complex setup as a reference.
+
+
+## Design Details
+
+TBD
+
+### Test Plan
+
+The current e2e tests will be changed to use the new TLS setup.
+
+### Upgrade / Downgrade Strategy
+
+For the most common setups we provide a migration script to the new config
+format. If only an old `keylime.conf` is found Keylime will error with a message
+that migration to the new format is required.
+
+Automatic downgrading will not be supported. The user can just use their old `keylime.conf`.
+
+
+### Dependency requirements
+
+
+
+## Drawbacks
+
+* Configuration must be migrated. We will not support the old and the new format
```

Comment:
Are we planning to provide a migration tool?

Reply:
Yes, but with the change on how the TLS configuration works, we only migrate when only one CA is used.

Reply:
Fair enough